infra/aws/terraform/artifacts.k8s.io/main.tf

```diff
@@ -0,0 +1,63 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// prefix prefixes every resource so that the resources
+// can be created without using the same names. Useful
+// for testing and staging
+
+variable "prefix" {
+  type        = string
+  default     = "test-"
+  description = "The prefix for all resources"
+
+  validation {
+    condition     = can(regex(".*-$|^$", var.prefix))
+    error_message = "The string must end with a hyphen or be empty."
+  }
+}
+
+
+module "eu-west-1" {
+  source = "./s3"
+
+  providers = {
+    aws = aws.eu-west-1
+  }
+
+  prefix = var.prefix
+}
+
+module "us-east-2" {
+  source = "./s3"
+
+  providers = {
+    aws = aws.us-east-2
+  }
+
+  prefix = var.prefix
+
+  s3_replication_iam_role_arn = "arn:aws:iam::513428760722:role/artifacts.k8s.io_s3writer"
+
+  s3_replication_rules = [
+    {
+      id                               = "artifacts-k8s-io-us-east-2-to-artifacts-k8s-io-us-west-2"
+      status                           = "Enabled"
+      priority                         = 2
+      destination_bucket_arn           = "arn:aws:s3:::${var.prefix}artifacts-k8s-io-us-west-2"
```

Comment:
This could use a tidy up.
For example, interpolate in the name of the source and target regions.

Reply:
I could use some help here.  So I'm focusing on test-artifacts.k8s.io at the moment.  I put the replication in #4714.  You can see that I was able to get some interpolation, but I still need to repeat the provider blocks.  So it's better, but still suboptimal.  If you know any tricks, happy to review PRs / test them out!